src/main/java/ch/njol/skript/sections/EffSecSendResourcePack.java

```diff
@@ -0,0 +1,214 @@
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.EffectSection;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.Trigger;
+import ch.njol.skript.lang.TriggerItem;
+import ch.njol.skript.variables.Variables;
+import ch.njol.util.Kleenean;
+import ch.njol.util.StringUtils;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import ch.njol.skript.util.Utils;
+import org.jetbrains.annotations.UnknownNullability;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@Name("Send Resource Pack")
+@Description({
+	"Request that the player's client download and switch resource packs. The client will download",
+	"the resource pack in the background, and will automatically switch to it once the download is complete.",
+	"The URL must be a direct download link.",
+	"",
+	"The [**resource pack request action**](events.html#resource_pack_request_action) can be used to check",
+	"the status of the sent resource pack request.",
+	"",
+	"Take a look at [**Resource Pack Values**](https://docs.skriptlang.org/effects.html#EffResourcePackValues) to see additional options."
+})
+@Examples({
+	"on join:",
+		"\tsend the resource pack from \"URL\" to the player",
+	"",
+		"\tsend the resource pack from \"URL\" to the player using:",
+			"\t\tset the resource pack uuid to \"1\"",
+			"\t\tset the resource pack hash to \"Hash\"",
+			"\t\tset the resource pack prompt to \"Please Download\"",
+			"\t\tforce the player to accept"
+})
+@Since("2.4, INSERT VERSION (section)")
+public class EffSecSendResourcePack extends EffectSection {
+
+	public static class ResourcePackEvent extends Event {
+		@Nullable
+		private String id, hash, prompt;
+		@Nullable
+		private Boolean force;
+
+		public ResourcePackEvent() {}
+
+		public void setId(String id) {
+			this.id = id;
+		}
+		public void setHash(String hash) {
+			this.hash = hash;
+		}
+		public void setPrompt(String prompt) {
+			this.prompt = prompt;
+		}
+		public void setForce(Boolean force) {
+			this.force = force;
+		}
+
+		public String getId() {
+			return id;
+		}
+		public String getHash() {
+			return hash;
+		}
+		public String getPrompt() {
+			return prompt;
+		}
+		public Boolean getForce() {
+			return force;
+		}
+
+
+		@Override
+		@NotNull
+		public HandlerList getHandlers() {
+			throw new IllegalStateException();
+		}
+	}
+
+	static {
+		Skript.registerSection(EffSecSendResourcePack.class,
+			"send [a|the] resource pack [at|from [[the] URL]] %string% to %players% [using|with]",
+			"send [a|the] resource pack [at|from [[the] URL]] %string% with hash %string% to %players%"
+			);
+
+	}
+
+	@UnknownNullability
+	@SuppressWarnings("null")
+	private Expression<String> url;
```

Comment:
I don't think the nullability of these is unknown. If it is though, it should at least be moved in-line, and the suppress ones should just be removed. Same goes for the ones below, i.e. `recipients`, `trigger`, `oldHash`